pkg/rbeconfigsgen/rbeconfigsgen.go

```diff
@@ -0,0 +1,847 @@
+package rbeconfigsgen
+
+import (
+	"archive/tar"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/coreos/go-semver/semver"
+)
+
+const (
+	buildHeader = `# Copyright 2020 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file is auto-generated by github.com/bazelbuild/bazel-toolchains/pkg/rbeconfigsgen
+# and should not be modified directly.
+`
+)
+
+var (
+	platformsToolchainBuildTemplate = template.Must(template.New("platformsBuild").Parse(buildHeader + `
+package(default_visibility = ["//visibility:public"])
+
+{{ if .CppToolchainTarget }}
+toolchain(
+    name = "cc-toolchain",
+    exec_compatible_with = [
+{{ range .ExecConstraints }}        "{{ . }}",
+{{ end }}    ],
+    target_compatible_with = [
+{{ range .TargetConstraints }}        "{{ . }}",
+{{ end }}    ],
+    toolchain = "{{ .CppToolchainTarget }}",
+    toolchain_type = "@bazel_tools//tools/cpp:toolchain_type",
+){{ end }}
+
+platform(
+    name = "platform",
+    parents = ["@local_config_platform//:host"],
+    constraint_values = [
+{{ range .ExecConstraints }}        "{{ . }}",
+{{ end }}    ],
+    remote_execution_properties = """
```

Comment:
Can you use here `exec_properties = create_rbe_exec_properties_dict(...)`?
See https://github.com/bazelbuild/bazel-toolchains/blob/master/rules/exec_properties/README.md#create_rbe_exec_properties_dict

Reply:
As discussed offline, I'm changing this to use a dictionary but stopping short of using create_rbe_exec_... to keep dependencies simple.